src/query/execute_statement.c

```diff
@@ -2874,7 +2945,7 @@ do_alter_serial (PARSER_CONTEXT * parser, PT_NODE * statement)
   obj_tmpl = NULL;
 
 end:
-  if (!OID_ISNULL (&serial_obj_id))
+  if (!OID_ISNULL (&serial_obj_id) && alter_serial_code == PT_SERIAL_OPTION)
```

Comment:
[Q] What does serial_decache() function do?

Reply:
The serial_decache() function only serves to free the cache values in the serial_cache_pool.
When the user calls nextval, it updates the serial_cache_pool and fetches the serial value in the cache_pool.

ex) When executing nextval after decache, current_val increases by the cache value.

`CREATE SERIAL cubrid_seq cache 5;`
`SELECT cubrid_seq.nextval FROM dual;`
> serial_current_value(cubrid_seq)
======================
 1  

`SELECT name, current_val, cached_num FROM db_serial WHERE unique_name='dba.cubrid_seq';`
>   name          |        current_val      |       cached_num
=========================================================
  'cubrid_seq'    |      5                       |        5

/* Use serial_decache() */
`ALTER SERIAL cubrid_seq MAXVALUE 30000;`
`SELECT name, current_val, cached_num FROM db_serial WHERE unique_name='dba.cubrid_seq';`

>   name       |      current_val       |     cached_num
=========================================================
  'cubrid_seq'      |        5               |                5

/* Update serial_cache_pool and get the serial value in cache_pool */
`SELECT cubrid_seq.nextval FROM dual;`
>   serial_next_value(cubrid_seq, 1)
======================
   6   

`SELECT name, current_val, cached_num FROM db_serial WHERE unique_name='dba.cubrid_seq';`
>   name            |      current_val     |       cached_num
=========================================================
  'cubrid_seq'      |    10                    |          5

Reply:
After serial_decache(), executing nextval calls the xserial_get_next_value_internal() function to update the serial_cache_pool.

> src/query/serial.c:xserial_get_next_value()

```
  ...
  /* used serial cache */
  entry = (SERIAL_CACHE_ENTRY *) mht_get (serial_Cache_pool.ht, oid_p);
  if (entry != NULL)
  {
	/* 
	 * If serail_cache_pool exists
	 * Get the serial value from cache_pool.
	 */
	...
	ret = serial_get_next_cached_value (thread_p, entry, num_alloc);
  }
  else
  {
	/* 
	 * If serail_cache_pool does not exist
	 * After creating serail_cache_pool, get the serial value from cache_pool.
	 */
	...
	ret = xserial_get_next_value_internal (thread_p, result_num, oid_p, num_alloc);
  }
  ...
```

Reply:
The Serial Decache logic is not related to adding SQL syntax, so we will create and address new issues.